Shallows

Kurfursten opened the pull request:
It's not the most elegant thing ever but it's better than our "completely cover their face" system of water.

![image](https://user-images.githubusercontent.com/9782036/104956626-240f1e80-5992-11eb-9280-69066958cbe0.png)

Not added to any maps so no CL

PrimeDSS13 commented:
god bless this has been needed for so long

Exxion commented on code/game/turfs/unsimulated/beach.dm:
```diff
@@ -15,15 +15,57 @@
 	water.plane = MOB_PLANE
-	overlays += water
+
+//enter is "trying to enter", we use this instead of Entered() because we don't want to count the entering object
```
What?

EDIT: Oh I read some more and figured it out
Couldn't you just use `Entered()` and subtract one from the count? `Enter()` shouldn't have side effects

Kurfursten commented:
- Addressed concerns
- Added a new "sandbar" tile. Sandbars are too shallow to even give you an overlay, and you can see the sand under. Good for borders like with #28623

![image](https://user-images.githubusercontent.com/9782036/105083202-36ed2600-5a5a-11eb-9119-b4f1395e6f85.png)

This PR must clear for #28618 because I want to use the shallows in the hotspring
